packages/driver/cypress/integration/commands/net_stubbing_spec.ts

```diff
@@ -1633,6 +1633,86 @@ describe('network stubbing', { retries: { runMode: 2, openMode: 0 } }, function
       cy.visit('/fixtures/utf8-post.html')
     })
 
+    // https://github.com/cypress-io/cypress/issues/16327
+    context('request url querystring', () => {
+      it('parse query correctly', () => {
+        cy.intercept({ url: '/users*' }, (req) => {
+          expect(req.query.someKey).to.deep.equal('someValue')
+          expect(req.query).to.deep.equal({ someKey: 'someValue' })
+        }).as('getUrl')
+
+        cy.window().then((win) => {
+          const xhr = new win.XMLHttpRequest()
+
+          xhr.open('GET', '/users?someKey=someValue')
+          xhr.send()
+        })
+
+        cy.wait('@getUrl')
+      })
+
+      context('reconcile changes', () => {
+        it('by assigning a new query parameter obj', () => {
+          cy.intercept({ url: '/users*' }, (req) => {
+            req.query = {
+              a: 'b',
+            }
+
+            expect(req.url).to.eq('http://localhost:3500/users?a=b')
+          }).as('getUrl')
+
+          cy.window().then((win) => {
+            const xhr = new win.XMLHttpRequest()
+
+            xhr.open('GET', '/users?someKey=someValue')
+            xhr.send()
+          })
+
+          cy.wait('@getUrl')
+        })
+
+        it('by setting new properties', () => {
+          cy.intercept({ url: '/users*' }, (req) => {
+            expect(req.query.a).to.eq('b')
+            req.query.c = 'd'
+
+            expect(req.url).to.eq('http://localhost:3500/users?a=b&c=d')
+          }).as('getUrl')
+
+          cy.window().then((win) => {
+            const xhr = new win.XMLHttpRequest()
+
+            xhr.open('GET', '/users?a=b')
+            xhr.send()
+          })
+
+          cy.wait('@getUrl')
+        })
+
+        it('by doing both', () => {
+          cy.intercept({ url: '/users*' }, (req) => {
+            req.query = {
+              a: 'b',
+            }
+
+            expect(req.query.a).to.eq('b')
+            req.query.c = 'd'
```

Comment:
a test for `delete req.query.x` at least would be nice, once the `Proxy` is more fleshed out

Reply:
Done.